src/app/group/timeline/timeline.component.ts

```diff
@@ -9,18 +9,45 @@ import { TASKS } from './mock-tasks'
 })
 
 export class TimelineComponent implements OnInit {
-  tasks = TASKS;
-  selectedTask: Task;
+  tasks = TASKS
+  selectedTask: Task
+  currentDate: Date = new Date()
+  fiveDaysInMillisec: number = 5 * 24 * 60 * 60 * 1000
 
   onSelect(currentTask: Task): void {
-    if(this.selectedTask === currentTask){
-      this.selectedTask = null;
-    } else {
-      this.selectedTask = currentTask;
+    this.selectedTask = currentTask
+  }
+
+  isOldTask(task: Task): boolean {
+    let taskDate: Date = new Date(task.date)
+
+    if (((this.currentDate.getTime() - taskDate.getTime()) > this.fiveDaysInMillisec)
+       && task != this.selectedTask) { //so that we don't try to paint tasks in blue and yellow simultaneously
+      return true
     }
   }
 
-  constructor() { }
+  constructor() {
+
+    let todayTime = new Date(this.currentDate.toLocaleDateString()).getTime()
+    let activeTask: Task = this.tasks[0];
+
+    for (let task of this.tasks) {
+      let taskTime = new Date(task.date).getTime()
+
+      if (todayTime == taskTime) {
+        this.selectedTask = task
```

Comment:
I t would be better to use strict equality operator?

Reply:
fixed